src/blockdata/script.rs

```diff
@@ -1117,6 +1117,18 @@ mod test {
         let key = PublicKey::from_str(&keystr[2..]).unwrap();
         script = script.push_key(&key); comp.extend(Vec::from_hex(keystr).unwrap().iter().cloned()); assert_eq!(&script[..], &comp[..]);
 
+        // XOnly keys
+        // Notice the "20" which prepends each keystr. That 20 is hexidecimal for "32". The Builder automatically adds the 32 opcode 
+        // to our script in order to give a heads up to the script compiler that it should add the next 32 bytes to the stack.
+        // To mimic the Builder's automatic insertion of the 32 opcode, we manually add it to the Vec and slice it out in the from_str.
+        let keystr = "209997a497d964fc1a62885b05a51166a65a90df00492c8d7cf61d6accf54803be"; // From: https://github.com/bitcoin-core/btcdeb/blob/e8c2750c4a4702768c52d15640ed03bf744d2601/doc/tapscript-example.md?plain=1#L43
```

Comment:
Can you make a separate test for this. In future, it might be confusing if we add `key` and `x_only_key` in the same commit

Reply:
Thanks for reviewing. I created a separate test and named it `script_x_only_keys()`/